vmware-base/src/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -1110,4 +1148,39 @@ public String getNetworkName(String netMorVal) throws Exception {
         }
         return networkName;
     }
+
+    public void createPortGroup(HostPortGroupSpec spec) throws Exception {
+        if (spec == null) {
+            return;
+        }
+
+        synchronized (_mor.getValue().intern()) {
+            HostNetworkSystemMO hostNetMo = getHostNetworkSystemMO();
+            if (hostNetMo == null) {
+                return;
+            }
+
+            ManagedObjectReference morNetwork = getNetworkMor(spec.getName());
+            if (morNetwork == null) {
+                hostNetMo.addPortGroup(spec);
+            }
+        }
+    }
+
+    public void copyPortGroupsFromHost(HostMO srcHost) throws Exception {
+        if (srcHost == null) {
+            return;
+        }
+
+        List<HostPortGroupSpec> portGroupSpecs = srcHost.getHostPortGroupSpecs();
+        if (portGroupSpecs == null || portGroupSpecs.isEmpty()) {
```

Comment:
Hi @sureshanaparti, could you please use CollectionUtils.isEmpty?
It checks if the List is empty, returning true in case of null. (https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html#isEmpty%28java.util.Collection%29)
Thanks.